src/libraries/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1979,6 +2004,14 @@ private int InternalGetHashCode(int hashCode)
             return hashCode & Lower31BitMask;
         }
 
+#if BIT64
+        private int FastMod(int value, int divisor)
```

Comment:
I suggest instead to extract `ref int GetBucket(uint hashCode)` same as in Dictionary. That puts it in one place, including the `#if BIT64` stuff. With `[MethodImpl(MethodImplOptions.AggressiveInlining)]` on it, as for Dictionary, it will inline and be just as efficient.